nemoguardrails/colang/v2_x/library/core.co

```diff
@@ -291,12 +291,31 @@ flow await_flow_by_name $flow_name $new_flow_instance_uid = None
   match FlowStarted(flow_id=$flow_name, flow_instance_uid=$new_flow_instance_uid) as $event_ref
   match $event_ref.flow.Finished()
 
-flow wait until done $ref
-  """Wait until an action or flow is done (has finished or was stopped)."""
+flow it finished $ref -> $finished_event_ref
+  """Wait until a flow or action has finished."""
 
-  if type($ref) != "Action" and type($ref) != "FlowState"
-    log "flow `is done` only supports action and flow references."
+  $type = type($ref)
+  if $type != "Action" and $type != "FlowState":
+    log "flow `it finished` expects a flow or action reference as parameter."
     abort
 
-  if $ref.status.value != "finished" and $ref.status.value != "stopped":
-    match $ref.Finished()
+  if $type == "FlowState" and $ref.status.value == "stopped":
```

Comment:
I think the type check here is not necessary unless you want it for documentation purposes